ebs-csi.tf

```diff
@@ -0,0 +1,71 @@
+module "kms_key_csi" {
+  count  = var.aws_ebs_csi_driver_enabled && var.ebs_csi_kms_key_arn == "" ? 1 : 0
+  source = "github.com/dedicatted/devops-tech//terraform/aws/modules/terraform-aws-kms"
+
+  # Add any necessary input variables for the kms module
+  # For example:
+  deletion_window_in_days = 7
+  key_administrators = [
+    data.aws_caller_identity.current.arn
+  ]
+  key_users = [
+    module.ebs_csi_irsa_role[0].iam_role_arn
+  ]
+}
+
+module "ebs_csi_irsa_role" {
+  count  = var.aws_ebs_csi_driver_enabled ? 1 : 0
+  source = "terraform-aws-modules/iam/aws//modules/iam-role-for-service-accounts-eks"
+
+  role_name             = var.ebs_csi_irsa_role_name
+  attach_ebs_csi_policy = true
+
+  oidc_providers = {
+    ex = {
+      provider_arn               = data.aws_eks_cluster.eks_cluster.identity[0].oidc[0].issuer
+      namespace_service_accounts = ["kube-system:ebs-csi-controller-sa"]
+    }
+  }
+}
+
+
+resource "helm_release" "ebs_csi_driver" {
+  count      = var.aws_ebs_csi_driver_enabled ? 1 : 0
+  name       = "aws-ebs-csi-driver"
```

Comment:
please add depends_on for previous resources

Reply:
done